service/history/mutableState.go

```diff
@@ -120,37 +120,38 @@ type (
 		GetChildExecutionInfo(int64) (*persistence.ChildExecutionInfo, bool)
 		GetChildExecutionInitiatedEvent(int64) (*workflow.HistoryEvent, bool)
 		GetCompletionEvent() (*workflow.HistoryEvent, bool)
+		GetDecisionInfo(int64) (*decisionInfo, bool)
 		GetStartEvent() (*workflow.HistoryEvent, bool)
 		GetCurrentBranch() []byte
 		GetCurrentVersion() int64
 		GetExecutionInfo() *persistence.WorkflowExecutionInfo
 		GetEventStoreVersion() int32
 		GetHistoryBuilder() *historyBuilder
-		GetInFlightDecisionTask() (*decisionInfo, bool)
+		GetInFlightDecision() (*decisionInfo, bool)
+		GetPendingDecision() (*decisionInfo, bool)
 		GetLastFirstEventID() int64
 		GetLastWriteVersion() int64
 		GetNextEventID() int64
 		GetPreviousStartedEventID() int64
-		GetPendingDecision(int64) (*decisionInfo, bool)
 		GetPendingActivityInfos() map[int64]*persistence.ActivityInfo
 		GetPendingTimerInfos() map[string]*persistence.TimerInfo
 		GetPendingChildExecutionInfos() map[int64]*persistence.ChildExecutionInfo
+		GetPendingRequestCancelInfos() map[int64]*persistence.RequestCancelInfo
+		GetPendingSignalInfos() map[int64]*persistence.SignalInfo
```

Comment:
Can we call it GetPendingSignalExternalInfos()? GetPendingSignalInfos sounds like getting received signals.

Reply:
sure